client/driver/docker.go

```diff
@@ -2183,3 +2180,25 @@ type createContainerClient interface {
 	ListContainers(docker.ListContainersOptions) ([]docker.APIContainers, error)
 	RemoveContainer(opts docker.RemoveContainerOptions) error
 }
+
+func parseDockerImage(image string) (repo, tag string) {
+	repo, tag = docker.ParseRepositoryTag(image)
+	if tag != "" {
+		return repo, tag
+	}
+	if i := strings.IndexRune(image, '@'); i > -1 { // Has digest (@sha256:...)
```

Comment:
is there a risk of someone using `@` in their repo name/url/tag? wonder if its more safe to actually sniff for `@sha256:` in its entirely - not aware of the docker repo spec if its invalid or not

Reply:
I'm pretty sure that '@' is not a valid repo name. The official spec is codified here:
https://github.com/docker/distribution/blob/master/reference/regexp.go